// @lemoncode/fonk-is-url-validator/dist/@lemoncode/fonk-is-url-validator.umd.js
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@lemoncode/fonk')) :
  typeof define === 'function' && define.amd ? define(['exports', '@lemoncode/fonk'], factory) :
  (global = global || self, factory(global['@lemoncode/fonk-is-url-validator'] = {}, global.Fonk));
}(this, function (exports, fonk) { 'use strict';

  var VALIDATOR_TYPE = 'IS_URL';
  var defaultMessage = 'Provided value is not a valid url';
  var setErrorMessage = function setErrorMessage(message) {
    return defaultMessage = message;
  };

  var isDefined = function isDefined(value) {
    return value !== void 0 && value !== null && value !== '';
  };

  var regex = /^(?:((?:(?:https?|ftp):\/\/))|(?:www.))+(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z0-9\u00a1-\uffff][a-z0-9\u00a1-\uffff_-]{0,62})?[a-z0-9\u00a1-\uffff]\.)+(?:[a-z\u00a1-\uffff]{2,}\.?))(?::\d{2,5})?(?:[/?#]\S*)?$/;

  var isUrl = function isUrl(url) {
    return regex.test(url);
  };

  var validator = function validator(fieldValidatorArgs) {
    var value = fieldValidatorArgs.value,
        _fieldValidatorArgs$m = fieldValidatorArgs.message,
        message = _fieldValidatorArgs$m === void 0 ? defaultMessage : _fieldValidatorArgs$m,
        customArgs = fieldValidatorArgs.customArgs;
    var succeeded = !isDefined(value) || isUrl(value);
    return {
      succeeded: succeeded,
      message: succeeded ? '' : fonk.parseMessageWithCustomArgs(message || defaultMessage, customArgs),
      type: VALIDATOR_TYPE
    };
  };

  var validator$1 = /*#__PURE__*/Object.freeze({
    __proto__: null,
    setErrorMessage: setErrorMessage,
    validator: validator
  });

  exports.isUrl = validator$1;

  Object.defineProperty(exports, '__esModule', { value: true });

}));
